### ISSUE TITLE
Beta Checklist!

### ISSUE THREAD

**Issue by tajmone** [tajmone/Git-Buch_EN#22]
We're about to enter the Beta stage of the book, where its HTML version will be made available on the WWW via the repository GHPages website — with a clear admonition regarding its Beta status, and a call for help to polish the translation before reaching v1.0.0.

Below is a checklist of all the things that need to be done before moving to the Beta stage.

- [ ] **GitHub Pages**:
    + [ ] Configure repository to serve the website from the `docs/` folder.
    + [ ] Add empty `docs/.nojekyll` file to enforce static website.
    + [ ] `master` branch — In order to use GitHub Pages, we'll need to publish the current book to the default branch (`master`).
        * [ ] We should stop using the `alpha-dev` branch, and create a new `beta-dev` branch for all PRs, which will be then merged into `master` — i.e. until we reach version 1.0.0 of the book. We shouldn't accept PRs directly on `master`.
- [ ] **Rake automation** — all Bash and batch scripts should be dropped in favor of a `Rakefile` providing cross-platform tasks to achieve the same goals:
    + [ ] Dia Diagrams — an optional build task, should check Dia version and that SVGO is available before executing.
    + [ ] HTML Book — needs to build into the `docs/` folder instead.
    + [ ] HTML Book Preview — build the HTML book (no URI-Data) in `docs_src/`, the HTML doc is then ignored by Git.
    + [ ] Git Hooks — installation and removal of the Git hooks become Rake tasks.
        * [ ] If we decide to keep the hook for building the HTML book, then it should invoke the Rake task, not use its own Shell script, to avoid conflicting output due to different AsciiDoc options.
    + [ ] EClint — validation of the repository files should also become a Rake task, so we can use Ruby to exclude from the result files which are ignored by Git. Also, we might want to have two separate tasks:
        * [ ] Validate whole repository — for local validation.
        * [ ] Validate commit files — to validate commits before pushing or creating a PR. This task should be assigned to the Git pre-commit hook.
- [ ] **Update Docs**:
    + [ ] Main `README.md`:
        * [ ] Add/fix badges:
            - [ ] Ruby v3.
            - [ ] Status Beta.
    + [ ] Add `CONTRIBUTING.md` — we'll need to provide some guidelines regarding the project standards, which branch to commit to, etc.

The above list is probably incomplete, and it will be updated as new pending chores come up during discussions and practical work. All suggestions are welcome.

**Comment by tajmone** [comment]
# Beta Stage Active!

@SicroAtGit, development has now moved from `alpha-dev` to `beta-dev`, and the book can be previewed by GHPages:

- https://git.io/git-book

Some of the above tasks have not yet been completed, but it's OK — I'll just move them to individual Issues to be dealt with in the Beta stage.

**Comment by tajmone** [comment]
# GitHub Community Announcement

I've announced the availability for the Beta preview, also asking for help with the proofreading and revision work:

https://github.community/t/the-git-book-free-english-translation-available-on-github/202390
